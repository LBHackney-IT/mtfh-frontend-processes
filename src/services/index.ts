export { default as locale } from "./locale";
export { default as menu } from "./menu";
export * as processes from "./processes";
export * from "./config";
