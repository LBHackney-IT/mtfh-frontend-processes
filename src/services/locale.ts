import reactHtmlParser from "react-html-parser";

import { AssetAddress } from "@mtfh/common/lib/api/asset/v1";

const locale = {
  backButton: "Back",
  bookAppointment: "Book Appointment",
  next: "Next",
  cancel: "Cancel",
  cancelProcess: "Cancel process",
  change: "Change",
  views: {
    soleToJoint: {
      title: "Sole tenant requests a joint tenure",
      steps: {
        selectTenant: "Select person",
        checkEligibility: "Eligibility Checks",
        supportingDocuments: "Supporting documents",
        finish: "Finish",
        breachOfTenancy: "Breach of tenure check",
        requestDocuments: "Request Documents",
        reviewDocuments: "Review Documents",
        submitCase: "Submit case",
      },
      actions: {
        reassignCase: "Reassign case",
        cancelProcess: "Cancel process",
        caseActivityHistory: "Case activity history",
      },
    },
    selectTenants: {
      selectTenantHint:
        "This person will be asked for proof of relationship e.g. marriage or civil partnership certificate.",
      selectTenantLabel: "Who do you want to add as a joint tenant?",
      addToTenureText: "If the person you want to add is not listed you must first",
      addToTenureLink: "add them to the tenure",
      noHouseholdMembersOver18:
        "There are no household members over the age of 18 associated to this.",
    },
    processesMenu: {
      selectProcessDescription: () =>
        reactHtmlParser(
          "<p>Start a process by selecting a process category below. <br /> Where a process is not yet supported in <strong>Manage My Home</strong> these links will open the appropriate interim Google form.</p>",
        ),
    },
    checkEligibility: {
      autoCheckIntro:
        "Manage My Home has used tenure, person and property information to automatically check eligibility for this application.",
      passedChecks: "Passed automatic eligibilty checks",
      failedChecks: "Not eligible for a sole to joint tenure",
    },
    breachOfTenancy: {
      autoCheckIntro:
        "Manage My Home has used tenure, person and property information to automatically check eligibility for this application.",
      passedChecks: "Eligibility checks passed",
      failedChecks: "Not eligible for a sole to joint tenure",
    },
    reviewDocuments: {
      passedChecks: "Eligibility checks passed",
      documentsRequested: "Documents requested",
      checkSupportingDocumentsAppointment:
        "I need to make an appointment to check supporting documents",
      appointmentScheduled: "Office appointment scheduled",
      seenPhotographicId:
        "I confirm that an outcome letter has been sent to the resident",
      seenSecondId:
        "I confirm I have seen a second form of ID (does not have to be photographic)",
      isNotInImmigrationControl:
        "I confirm that the prospective tenant is not subject to immigration control under the Asylum And Immigration Act 1996",
      seenProofOfRelationship:
        "I confirm I have seen proof of relationship to the existing tenant",
      incomingTenantLivingInProperty:
        "I confirm I have seen 3 separate documents proving the proposed tenant has been living at the property for a minimum of 12 months",
    },
  },
  errors: {
    unableToFetchRecord: "There was a problem retrieving the record",
    unableToFetchRecordDescription:
      "Please try again. If the issue persists, please contact support.",
    unableToFindProcess: "Invalid Process",
    unableToFindProcessDescription: "We have been unable to find a process of this type.",
    unableToFindState: "Invalid State",
    unableToFindStateDescription: "We have been unable to process a state of this type.",
  },
  title: "Processes",
  loadingText: "Submitting...",
  components: {
    startProcess: {
      buttonLabel: "Start process",
      thirdPartyHeading: "Sharing information with third parties",
    },
    entitySummary: {
      tenurePaymentRef: "Tenure payment ref",
      address: (assetAddress: AssetAddress): string => {
        const {
          postPreamble,
          addressLine1,
          addressLine2,
          addressLine3,
          addressLine4,
          postCode,
        } = assetAddress;

        return [
          postPreamble,
          addressLine1,
          addressLine2,
          addressLine3,

          addressLine4,
          postCode,
        ]
          .filter((addressLine) => !!addressLine)
          .join(" ");
      },
    },
  },
};

export default locale;
