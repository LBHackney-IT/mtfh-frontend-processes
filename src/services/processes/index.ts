export * from "./sole-to-joint";
export * from "./change-of-name";
