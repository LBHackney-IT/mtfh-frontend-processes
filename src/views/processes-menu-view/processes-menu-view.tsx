import { Link as RouterLink, useParams } from "react-router-dom";

import { ProcessMenu } from "../../components";
import { locale } from "../../services";
import { TargetType } from "../../types";

import { Heading, Layout, Link } from "@mtfh/common/lib/components";

import "./styles.scss";

const { views, title, backButton } = locale;
const { processesMenu } = views;

export const ProcessesMenuView = ({ targetType }: { targetType: TargetType }) => {
  const { id } = useParams<{ id: string }>();

  return (
    <Layout data-testid="processes-menu" className="mtfh-processes-menu">
      <Link as={RouterLink} to={`/${targetType}/${id}`} variant="back-link">
        {backButton}
      </Link>
      <Heading as="h1">{title}</Heading>
      <div className="mtfh-processes-menu__description">
        {processesMenu.selectProcessDescription()}
      </div>
      <ProcessMenu id={id} targetType={targetType} />
    </Layout>
  );
};
