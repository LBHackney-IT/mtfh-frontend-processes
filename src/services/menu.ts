import * as processes from "./processes";

import { Asset } from "@mtfh/common/lib/api/asset/v1/types";
import { Tenure } from "@mtfh/common/lib/api/tenure/v1/types";

interface MenuProps {
  label: string;
  link: string;
  processes?: { label: string; link: string; targetType: string }[];
  getPplQuery?: (context: any) => string;
}

const getProcessLink = (processName: string) => {
  const process = processes[processName];
  return {
    targetType: process.targetType,
    label: process.title,
    link: `/processes/${process.processName}/start`,
  };
};

const getPplQueryForRequestALegalReferral = ({ tenure, asset }): string => {
  if (!tenure || !asset) {
    return "";
  }
  const {
    householdMembers,
    tenuredAsset: { propertyReference },
  } = tenure as Tenure;
  const {
    assetAddress: { addressLine1, postCode },
  } = asset as Asset;
  const tenantNames = householdMembers
    .filter((member) => member.isResponsible)
    .map((member) => member.fullName)
    .join(", ");
  const address = addressLine1;
  const query = `usp=pp_url&entry.1699199605=${tenantNames}&entry.2086190845=${address}&entry.1956036523=${postCode}&entry.1140527432=${propertyReference}`;
  const encodedQuery = encodeURI(query);
  return encodedQuery;
};

const getPplQueryTenancyChange = ({ tenure }): string => {
  if (!tenure) {
    return "";
  }
  const {
    householdMembers,
    paymentReference,
    tenuredAsset: { fullAddress, propertyReference },
  } = tenure as Tenure;

  const tenant = householdMembers.find((member) => member.isResponsible);
  const query = `usp=pp_url&entry.1942863541=${tenant?.fullName}&entry.1090481808=${fullAddress}&entry.321634337=${propertyReference}&entry.582205322=${paymentReference}`;
  const encodedQuery = encodeURI(query);
  return encodedQuery;
};

const menu: MenuProps[] = [
  {
    label: "Sole to Joint",
    link: "https://docs.google.com/forms/d/e/1FAIpQLSdgJ9DSgGI0Aj7GO1bzLbbrArPabjS8DQwmvwb9ltB-qYYESA/viewform",
    processes: [
      getProcessLink(processes.soletojoint.processName),
      getProcessLink(processes.changeOfName.processName),
    ],
  },
  {
    label: "Other tenancy changes",
    link: "https://docs.google.com/forms/d/e/1FAIpQLSdgJ9DSgGI0Aj7GO1bzLbbrArPabjS8DQwmvwb9ltB-qYYESA/viewform",
    getPplQuery: getPplQueryTenancyChange,
  },
  {
    label: "Record a tenancy / audit homecheck",
    link: "https://docs.google.com/forms/d/e/1FAIpQLSfyXscVvgCBhbBBe7ADpV6cUqJoKrrTbU0YYRB_uXS2J4AHMQ/viewform",
  },
  {
    label: "Find legal referral response",
    link: "https://docs.google.com/spreadsheets/d/1fd6FwBXvBo2DmX8qZaXUr7hX9sE6Zr7HZsPlA_OO8Hs/edit#gid=1251146140",
  },
  {
    label: "Request a legal referral",
    link: "https://docs.google.com/forms/d/e/1FAIpQLScpHsoJQgq112HIMbW06YAFpggijwQuODJM483HGbX_PMfCMw/viewform",
    getPplQuery: getPplQueryForRequestALegalReferral,
  },
  {
    label: "Record notice served",
    link: "https://docs.google.com/forms/d/e/1FAIpQLSfRSMia4FeT6WPjrmRlsLpOXyJ4tFs3Sv_O0toxUbv_FmIjKw/viewform",
  },
  {
    label: "Rehousing",
    link: "https://docs.google.com/forms/d/e/1FAIpQLScPXSs4-xxTH2N5dUwtfmEl5mpu7kxZ6mHe86Qy_rlDNu-GbA/viewform",
  },
  {
    label: "Person",
    link: "https://docs.google.com/forms/d/e/1FAIpQLScxrWFLF7M7A3B1yzY-LZFXkV0P0qv6igcQ4wEgWHQk8DaUCw/viewform",
  },
  {
    label: "Report Japanese knotweed",
    link: "https://docs.google.com/forms/d/e/1FAIpQLSceU2H-VWi2ewaEiOAkOWULVFU2DoPSen8SRPyJrmQtRpEhfQ/viewform?gxids=7628",
  },
  {
    label: "Report overgrown tree",
    link: "https://docs.google.com/forms/d/e/1FAIpQLSfLmlLraDyOfqYhdjkOfkNWWlHUUep05G5n0GN-ewizN0Fbuw/viewform?gxids=7628",
  },
  {
    label: "Report overgrown garden",
    link: "https://docs.google.com/forms/d/e/1FAIpQLSc3mOUUEQhdRiDsS0k9XDWlfGJG0CEFAC2rnC1sYmM-idt-Nw/viewform?gxids=7628",
  },
];

export default menu;
