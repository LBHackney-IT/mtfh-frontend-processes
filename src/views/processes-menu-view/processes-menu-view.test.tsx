import { render } from "@hackney/mtfh-test-utils";
import { screen } from "@testing-library/react";

import { locale } from "../../services";
import { ProcessesMenuView } from "./processes-menu-view";

import { $configuration } from "@mtfh/common/lib/configuration";

test("it renders processes menu view correctly", () => {
  const { container } = render(<ProcessesMenuView targetType="tenure" />, {
    url: "/processes/tenure/1234",
    path: "/processes/:targetType/:id",
  });
  expect(screen.getByText(locale.backButton)).toBeInTheDocument();
  expect(screen.getByText(locale.title)).toBeInTheDocument();
  expect(container).toMatchSnapshot();
});

describe("feature toggle on", () => {
  beforeEach(() => {
    const features = $configuration.getValue();
    $configuration.next({
      MMH: {
        ...features.MMH,
        featureToggles: { EnhancedProcessMenu: true },
      },
    });
  });

  test("it renders processes menu view correctly", () => {
    const { container } = render(<ProcessesMenuView targetType="tenure" />, {
      url: "/processes/tenure/1234",
      path: "/processes/:targetType/:id",
    });
    expect(screen.getByText(locale.backButton)).toBeInTheDocument();
    expect(screen.getByText(locale.title)).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
