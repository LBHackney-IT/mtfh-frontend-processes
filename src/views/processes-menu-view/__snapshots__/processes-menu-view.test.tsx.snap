// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`feature toggle on it renders processes menu view correctly 1`] = `
<div>
  <div
    class="mtfh-layout mtfh-layout--narrow mtfh-processes-menu"
    data-testid="processes-menu"
  >
    <div
      id="content"
    />
    <div
      class="mtfh-layout__container"
    >
      <div
        class="mtfh-layout__main"
      >
        <a
          class="govuk-back-link lbh-back-link lbh-link--no-visited-state"
          href="/tenure/1234"
        >
          Back
        </a>
        <h1
          class="lbh-heading-h1"
        >
          Processes
        </h1>
        <div
          class="mtfh-processes-menu__description"
        >
          <p>
            Start a process by selecting a process category below. 
            <br />
             Where a process is not yet supported in 
            <strong>
              Manage My Home
            </strong>
             these links will open the appropriate interim Google form.
          </p>
        </div>
        <ul
          class="mtfh-processes-menu__list"
        >
          <li>
            <details
              class="govuk-details lbh-details"
              data-module="govuk-details"
              data-testid="mtfh-details"
              id="mtfh-details"
            >
              <summary
                class="govuk-details__summary"
              >
                <span
                  class="govuk-details__summary-text"
                >
                  Change to a tenancy
                </span>
              </summary>
              <div
                class="govuk-details__text"
              >
                <ul
                  class="mtfh-processes-menu__process-list"
                >
                  <li>
                    <a
                      class="govuk-link lbh-link lbh-link--no-visited-state"
                      href="/processes/soletojoint/start/tenure/1234"
                    >
                      Sole tenant requests a joint tenure
                    </a>
                  </li>
                </ul>
              </div>
            </details>
          </li>
          <li>
            <a
              class="govuk-link lbh-link"
              href="https://docs.google.com/forms/d/e/1FAIpQLSdgJ9DSgGI0Aj7GO1bzLbbrArPabjS8DQwmvwb9ltB-qYYESA/viewform"
              rel="noopener noreferrer"
              target="_blank"
            >
              Other tenancy changes
            </a>
          </li>
          <li>
            <a
              class="govuk-link lbh-link"
              href="https://docs.google.com/forms/d/e/1FAIpQLSfyXscVvgCBhbBBe7ADpV6cUqJoKrrTbU0YYRB_uXS2J4AHMQ/viewform"
              rel="noopener noreferrer"
              target="_blank"
            >
              Record a tenancy / audit homecheck
            </a>
          </li>
          <li>
            <a
              class="govuk-link lbh-link"
              href="https://docs.google.com/spreadsheets/d/1fd6FwBXvBo2DmX8qZaXUr7hX9sE6Zr7HZsPlA_OO8Hs/edit#gid=1251146140"
              rel="noopener noreferrer"
              target="_blank"
            >
              Find legal referral response
            </a>
          </li>
          <li>
            <a
              class="govuk-link lbh-link"
              href="https://docs.google.com/forms/d/e/1FAIpQLScpHsoJQgq112HIMbW06YAFpggijwQuODJM483HGbX_PMfCMw/viewform"
              rel="noopener noreferrer"
              target="_blank"
            >
              Request a legal referral
            </a>
          </li>
          <li>
            <a
              class="govuk-link lbh-link"
              href="https://docs.google.com/forms/d/e/1FAIpQLSfRSMia4FeT6WPjrmRlsLpOXyJ4tFs3Sv_O0toxUbv_FmIjKw/viewform"
              rel="noopener noreferrer"
              target="_blank"
            >
              Record notice served
            </a>
          </li>
          <li>
            <a
              class="govuk-link lbh-link"
              href="https://docs.google.com/forms/d/e/1FAIpQLScPXSs4-xxTH2N5dUwtfmEl5mpu7kxZ6mHe86Qy_rlDNu-GbA/viewform"
              rel="noopener noreferrer"
              target="_blank"
            >
              Rehousing
            </a>
          </li>
          <li>
            <a
              class="govuk-link lbh-link"
              href="https://docs.google.com/forms/d/e/1FAIpQLScxrWFLF7M7A3B1yzY-LZFXkV0P0qv6igcQ4wEgWHQk8DaUCw/viewform"
              rel="noopener noreferrer"
              target="_blank"
            >
              Resident contact log
            </a>
          </li>
          <li>
            <a
              class="govuk-link lbh-link"
              href="https://docs.google.com/forms/d/e/1FAIpQLSceU2H-VWi2ewaEiOAkOWULVFU2DoPSen8SRPyJrmQtRpEhfQ/viewform?gxids=7628"
              rel="noopener noreferrer"
              target="_blank"
            >
              Report Japanese knotweed
            </a>
          </li>
          <li>
            <a
              class="govuk-link lbh-link"
              href="https://docs.google.com/forms/d/e/1FAIpQLSfLmlLraDyOfqYhdjkOfkNWWlHUUep05G5n0GN-ewizN0Fbuw/viewform?gxids=7628"
              rel="noopener noreferrer"
              target="_blank"
            >
              Report overgrown tree
            </a>
          </li>
          <li>
            <a
              class="govuk-link lbh-link"
              href="https://docs.google.com/forms/d/e/1FAIpQLSc3mOUUEQhdRiDsS0k9XDWlfGJG0CEFAC2rnC1sYmM-idt-Nw/viewform?gxids=7628"
              rel="noopener noreferrer"
              target="_blank"
            >
              Report overgrown garden
            </a>
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;

exports[`it renders processes menu view correctly 1`] = `
<div>
  <div
    class="mtfh-layout mtfh-layout--narrow mtfh-processes-menu"
    data-testid="processes-menu"
  >
    <div
      id="content"
    />
    <div
      class="mtfh-layout__container"
    >
      <div
        class="mtfh-layout__main"
      >
        <a
          class="govuk-back-link lbh-back-link lbh-link--no-visited-state"
          href="/tenure/1234"
        >
          Back
        </a>
        <h1
          class="lbh-heading-h1"
        >
          Processes
        </h1>
        <div
          class="mtfh-processes-menu__description"
        >
          <p>
            Start a process by selecting a process category below. 
            <br />
             Where a process is not yet supported in 
            <strong>
              Manage My Home
            </strong>
             these links will open the appropriate interim Google form.
          </p>
        </div>
        <ul
          class="mtfh-processes-menu__list"
        >
          <li>
            <a
              class="govuk-link lbh-link"
              href="https://docs.google.com/forms/d/e/1FAIpQLSdgJ9DSgGI0Aj7GO1bzLbbrArPabjS8DQwmvwb9ltB-qYYESA/viewform"
              rel="noopener noreferrer"
              target="_blank"
            >
              Sole to Joint
            </a>
          </li>
          <li>
            <a
              class="govuk-link lbh-link"
              href="https://docs.google.com/forms/d/e/1FAIpQLSdgJ9DSgGI0Aj7GO1bzLbbrArPabjS8DQwmvwb9ltB-qYYESA/viewform"
              rel="noopener noreferrer"
              target="_blank"
            >
              Other tenancy changes
            </a>
          </li>
          <li>
            <a
              class="govuk-link lbh-link"
              href="https://docs.google.com/forms/d/e/1FAIpQLSfyXscVvgCBhbBBe7ADpV6cUqJoKrrTbU0YYRB_uXS2J4AHMQ/viewform"
              rel="noopener noreferrer"
              target="_blank"
            >
              Record a tenancy / audit homecheck
            </a>
          </li>
          <li>
            <a
              class="govuk-link lbh-link"
              href="https://docs.google.com/spreadsheets/d/1fd6FwBXvBo2DmX8qZaXUr7hX9sE6Zr7HZsPlA_OO8Hs/edit#gid=1251146140"
              rel="noopener noreferrer"
              target="_blank"
            >
              Find legal referral response
            </a>
          </li>
          <li>
            <a
              class="govuk-link lbh-link"
              href="https://docs.google.com/forms/d/e/1FAIpQLScpHsoJQgq112HIMbW06YAFpggijwQuODJM483HGbX_PMfCMw/viewform"
              rel="noopener noreferrer"
              target="_blank"
            >
              Request a legal referral
            </a>
          </li>
          <li>
            <a
              class="govuk-link lbh-link"
              href="https://docs.google.com/forms/d/e/1FAIpQLSfRSMia4FeT6WPjrmRlsLpOXyJ4tFs3Sv_O0toxUbv_FmIjKw/viewform"
              rel="noopener noreferrer"
              target="_blank"
            >
              Record notice served
            </a>
          </li>
          <li>
            <a
              class="govuk-link lbh-link"
              href="https://docs.google.com/forms/d/e/1FAIpQLScPXSs4-xxTH2N5dUwtfmEl5mpu7kxZ6mHe86Qy_rlDNu-GbA/viewform"
              rel="noopener noreferrer"
              target="_blank"
            >
              Rehousing
            </a>
          </li>
          <li>
            <a
              class="govuk-link lbh-link"
              href="https://docs.google.com/forms/d/e/1FAIpQLScxrWFLF7M7A3B1yzY-LZFXkV0P0qv6igcQ4wEgWHQk8DaUCw/viewform"
              rel="noopener noreferrer"
              target="_blank"
            >
              Resident contact log
            </a>
          </li>
          <li>
            <a
              class="govuk-link lbh-link"
              href="https://docs.google.com/forms/d/e/1FAIpQLSceU2H-VWi2ewaEiOAkOWULVFU2DoPSen8SRPyJrmQtRpEhfQ/viewform?gxids=7628"
              rel="noopener noreferrer"
              target="_blank"
            >
              Report Japanese knotweed
            </a>
          </li>
          <li>
            <a
              class="govuk-link lbh-link"
              href="https://docs.google.com/forms/d/e/1FAIpQLSfLmlLraDyOfqYhdjkOfkNWWlHUUep05G5n0GN-ewizN0Fbuw/viewform?gxids=7628"
              rel="noopener noreferrer"
              target="_blank"
            >
              Report overgrown tree
            </a>
          </li>
          <li>
            <a
              class="govuk-link lbh-link"
              href="https://docs.google.com/forms/d/e/1FAIpQLSc3mOUUEQhdRiDsS0k9XDWlfGJG0CEFAC2rnC1sYmM-idt-Nw/viewform?gxids=7628"
              rel="noopener noreferrer"
              target="_blank"
            >
              Report overgrown garden
            </a>
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;
